src/extensions/Session.ts

```diff
@@ -0,0 +1,55 @@
+/**
+ * SPDX-FileCopyrightText: 2025 Nextcloud GmbH and Nextcloud contributors
+ * SPDX-License-Identifier: AGPL-3.0-or-later
+ */
+
+import { Extension } from '@tiptap/core'
+
+declare module '@tiptap/core' {
+	interface Commands<ReturnType> {
+		session: {
+			setSession: (session: {
+				documentId: number
+				id: number
+				token: string
+			}) => ReturnType
+			clearSession: () => ReturnType
+		}
+	}
+}
+
+interface StoredSession {
+	documentId: number
+	sessionId: number
+	sessionToken: string
+}
+
+const emptySession = {
+	documentId: 0,
+	sessionId: 0,
+	sessionToken: '',
+}
+
+export const Session = Extension.create({
+	name: 'session',
+
+	addStorage(): StoredSession {
+		return { ...emptySession }
+	},
```

Comment:
oh no... this will require tiptap v3 as previously the extension storage is a singleton and therefor used by all loaded editor instances at the same time. ( See https://github.com/ueberdosis/tiptap/pull/6060 )